switools/signaturelib.py

```diff
@@ -12,4 +14,45 @@ def getSigFileName( swiFile ):
       return SWIX_SIG_FILE_NAME
    return SWI_SIG_FILE_NAME
 
+def runCmd( cmd, workDir ):
+   try:
+      subprocess.check_call( cmd.split(" "), cwd=workDir )
+   except subprocess.CalledProcessError:
+      return False
+   return True
 
+def getOptimizations( swi, workDir ):
+   # unzip spits warnings when extracting non-existant files, so check first :-(
+   cmd = "unzip -Z1 %s" % swi
+   if not "swimSqshMap" in str( subprocess.check_output( cmd.split(" ") ) ):
+      return None # legacy image
+   cmd = "unzip -qq -o %s swimSqshMap" % os.path.abspath( swi )
+   try:
+      subprocess.check_call( cmd.split(" "), cwd=workDir )
+   except subprocess.CalledProcessError:
+      return None # legacy image
+
+   optims = []
+   with open( "%s/swimSqshMap" % workDir ) as f:
+      for line in f:
+         optim, _ = line.split( "=", 1 )
+         optims.append( optim )
+   return optims
+
+def extractSwadapt( swi, workDir ):
+   cmd = "unzip -o -qq %s swadapt" % os.path.abspath( swi )
+   return runCmd( cmd, workDir )
+
+def checkIsSwiFile( swi, workDir ):
+   if not os.path.isfile( swi ):
+      return False
+   # unzip spits warnings when extracting non-existant files, so check first :-(
+   cmd = "unzip -Z1 %s" % swi
```

Comment:
`zipfile` again.

Reply:
I was going to drop this comment, but then I can't see where you use `version` after extracting it, so maybe `zipfile` would be simpler here?